android-mvvm-core/src/main/java/com/manaschaudhari/android_mvvm/FieldUtils.java

```diff
@@ -0,0 +1,275 @@
+                    @Override
+                    public void onPropertyChanged(android.databinding.Observable observable, int i) {
+                        subscriber.onNext(field.get());
+                    }
+                };
+                field.addOnPropertyChangedCallback(callback);
+                subscriber.setCancellable(new Cancellable() {
+                    @Override
+                    public void cancel() throws Exception {
+                        field.removeOnPropertyChangedCallback(callback);
+                    }
+                });
+            }
+        });
+    }
+
+    @NonNull
+    public static <T> Observable<List<T>> toObservable(@NonNull final ObservableList<T> field) {
+        return Observable.create(new ObservableOnSubscribe<List<T>>() {
+            @Override
+            public void subscribe(final ObservableEmitter<List<T>> subscriber) {
+                subscriber.onNext(field);
+                final ObservableList.OnListChangedCallback<ObservableList<T>> callback = new ObservableList.OnListChangedCallback<ObservableList<T>>() {
+                    @Override
+                    public void onChanged(ObservableList observableList) {
+                        subscriber.onNext(field);
+                    }
+
+                    @Override
+                    public void onItemRangeChanged(ObservableList observableList, int i, int i1) {
+                        onChanged(observableList);
+                    }
+
+                    @Override
+                    public void onItemRangeInserted(ObservableList observableList, int i, int i1) {
+                        onChanged(observableList);
+                    }
+
+                    @Override
+                    public void onItemRangeMoved(ObservableList observableList, int i, int i1, int i2) {
+                        onChanged(observableList);
+                    }
+
+                    @Override
+                    public void onItemRangeRemoved(ObservableList observableList, int i, int i1) {
+                        onChanged(observableList);
+                    }
+                };
+                field.addOnListChangedCallback(callback);
+                subscriber.setCancellable(new Cancellable() {
+                    @Override
+                    public void cancel() throws Exception {
+                        field.removeOnListChangedCallback(callback);
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * A convenient wrapper for {@code ReadOnlyField#create(Observable)}
+     *
+     * @return DataBinding field created from the specified Observable
+     */
+    @NonNull
+    public static <T> ReadOnlyField<T> toField(@NonNull final Observable<T> observable) {
+        return ReadOnlyField.create(observable);
+    }
+
+    @NonNull
+    public static <T> Disposable bindTo(@NonNull final Observable<T> observable, @NonNull final ObservableField<T> field) {
```

Comment:
I think `bindTo` is dangerous. If a field is bound to multiple `Observable`s, the value can match to either of the Observables, which brings back mutation problems.

Lets look at usecases for `bindTo` and see if they can be tackled with `ReadOnlyField`?